pkg/endpoint/endpoint.go

```diff
@@ -1797,3 +1802,71 @@ func (e *Endpoint) identityLabelsChanged(owner Owner, myChangeRev int) error {
 
 	return nil
 }
+
+// setPolicyRevision sets the policy wantedRev with the given revision.
+func (e *Endpoint) setPolicyRevision(rev uint64) {
+	e.policyRevision = rev
+	k := 0
+	for i, w := range e.policyRevisionSignals {
+		select {
+		case <-w.ctx.Done():
+			close(w.ch)
+		default:
+			if rev >= w.wantedRev {
+				close(w.ch)
+			} else {
```

Comment:
Isn't this the wrong way around? When we have the desired revision, we close the channel and need to clean up. Until then the policySignal should stay in the list. Am I missing some neat magic? (we shorten the list at the end of the function)

Reply:
oh, I get the logic. It is difficult (but quite slick)

Reply:
This logic is not clear at first glance, please add comments indicating what it's doing.